resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ContentDescriptionManager.java

```diff
@@ -480,6 +482,37 @@ public synchronized void invalidateCache(boolean flush, IProject project) {
 		}
 	}
 
+	private static String getCurrentPlatformState() {
+		ResourcesPlugin plugin = ResourcesPlugin.getPlugin();
+		if (plugin == null) {
+			return EMPTY;
+		}
+		BundleContext bundleContext = plugin.getBundle().getBundleContext();
+		if (bundleContext == null) {
+			return EMPTY;
+		}
+		String ID = Arrays.stream(bundleContext.getBundles())
+				.map(bundle -> bundle.getBundleId() + SPACE + bundle.getSymbolicName() + SPACE + bundle.getVersion())
+				.sorted().collect(Collectors.joining(DELIMITER));
+		try {
+			MessageDigest digest = MessageDigest.getInstance("SHA-256"); //$NON-NLS-1$
+			return HexFormat.of().formatHex(digest.digest(ID.getBytes(StandardCharsets.UTF_8)));
+		} catch (NoSuchAlgorithmException e) {
+			return Integer.toHexString(ID.hashCode());
+		}
+	}
+
+	/**
+	 * @return the cached platform state, only public for testing purpose!
+	 */
+	public String getCachedPlatformState() {
```

Comment:
Currently there is not test requiring this, so probably this method can be private?

Reply:
yes I missed to revert that...